File: Components/WinForms/VB/PhotoViewer.MDI/Controls/ProgressControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors

Namespace PhotoViewer
	Partial Public Class ProgressControl
		Inherits XtraUserControl
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Property Maximum() As Integer
			Get
				Return progressBar.Properties.Maximum
			End Get
			Set(ByVal value As Integer)
				progressBar.Properties.Maximum = value
			End Set
		End Property
		Public Property Value() As Integer
			Get
				Return CInt(Fix(progressBar.EditValue))
			End Get
			Set(ByVal value As Integer)
				progressBar.EditValue = value
			End Set
		End Property
		Public Property ProgressText() As String
			Get
				Return label.Text
			End Get
			Set(ByVal value As String)
				label.Text = value
			End Set
		End Property
	End Class
End Namespace

File: Components/WinForms/CS/PhotoViewer.MDI/Controls/ProgressControl.cs
using DevExpress.XtraEditors;

namespace PhotoViewer {
    public partial class ProgressControl : XtraUserControl {
        public ProgressControl() {
            InitializeComponent();
        }
        public int Maximum {
            get { return progressBar.Properties.Maximum; }
            set { progressBar.Properties.Maximum = value; }
        }
        public int Value {
            get { return (int)progressBar.EditValue; }
            set { progressBar.EditValue = value; }
        }
        public string ProgressText {
            get { return label.Text; }
            set { label.Text = value; }
        }
    }
}
